File: msbuildservice/PlatformCI.MsBuildService.Models/Implementation/FilesystemMetadata.cs
using System;
using PlatformCI.MsBuildService.Models.Enums;

namespace PlatformCI.MsBuildService.Models.Implementation
{
    [Serializable]
    public class FilesystemMetadata
    {
        public string Path { get; set; }
        public FilesystemObjectType Type { get; set; }
        public string OptionalChangeId { get; set; }
    }
}

File: models/filesysmodel/filesystemmetadata.go
package filesysmodel

// object type of filesystem item
type FileSystemObjectType int

// file & folder types
const (
	FileType   FileSystemObjectType = 0
	FolderType FileSystemObjectType = 1
)

// Metadata describing the Path, Type, and Commit id of a file in the
type FileSystemMetadata struct {
	Path             string
	Type             FileSystemObjectType
	OptionalChangeId string
}
